extension/observer/ecsobserver/exporter.go

```diff
@@ -14,10 +14,115 @@
 
 package ecsobserver
 
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+)
+
+const (
+	defaultMetricsPath = "/metrics"
+)
+
 // CommonExporterConfig should be embedded into filter config.
 // They set labels like job, metrics_path etc. that can override prometheus default.
 type CommonExporterConfig struct {
 	JobName      string `mapstructure:"job_name" yaml:"job_name"`
 	MetricsPath  string `mapstructure:"metrics_path" yaml:"metrics_path"`
 	MetricsPorts []int  `mapstructure:"metrics_ports" yaml:"metrics_ports"`
 }
+
+type TaskExporter struct {
+	logger  *zap.Logger
+	cluster string
+}
+
+func NewTaskExporter(logger *zap.Logger, cluster string) *TaskExporter {
```

Comment:
Does this need to be public?

Reply:
Actually all the methods and struct in this package can be private because this is a component and won't be imported by other packages. I can change the factory functions to private, though I still prefer the struct and struct method in the 'public' form to indicate they should be used by other files within same package.

Reply:
Fixed in latest commit.

Reply:
If the structs also aren't used outside the package, please make them private - we don't want to expose anything to other packages if we don't have to.

Reply:
done, I will update other files later when most pending PRs are merged to avoid conflict between PRs.